### PR TITLE
Raising an exception when a schema is missing, added an option to migrate_schemas to ignore this error

### PR DESCRIPTION
**Pull request by mikicz**
So, I encountered a problem a while back, I must have deleted an schema on my local machine and I tried to migrate some new fields to my project and I got an weird error

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/home/miki/oss/tenant_virtualenv/tenant/tenant_whole/tenant_schemas/management/commands/migrate_schemas.py", line 51, in handle
    self.run_migrations(tenant.schema_name, settings.TENANT_APPS)
  File "/home/miki/oss/tenant_virtualenv/tenant/tenant_whole/tenant_schemas/management/commands/migrate_schemas.py", line 68, in run_migrations
    command.execute(*self.args, **self.options)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 89, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 20, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 49, in __init__
    self.build_graph()
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 176, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 65, in applied_migrations
    self.ensure_schema()
  File "/home/miki/oss/tenant_virtualenv/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 59, in ensure_schema
    raise MigrationSchemaMissing("Unable to create the django_migrations table (%s)" % exc)
django.db.migrations.exceptions.MigrationSchemaMissing: Unable to create the django_migrations table (relation "django_migrations" already exists
)
```

I was like, wtf, it took me a while to figure out. So now, I added a checkup to migrate_schemas, that the schema actually exists in the database and that raises an Exception otherwise. Also added an option to the command and wrote a little bit about it to the documentation.

### CODE REVIEW

**Comment by bernardopires** [comment]
Thanks for the patch. When would this be useful though? Why would you want to ignore this? Clearly if the schema is missing your database is an error state. If you deleted a schema you delete the corresponding row in the schemas table as well.

**Comment by mikicz** [comment]
Well I think the main point is having the exception at all, but I would find the ignoring usefull as well - for example if you have a client who stops paying for the service and you want to keep record you had that client, but don't want to keep the clients data.

**Comment by bernardopires** [comment]
In that case you should use a separate table for keeping that data. The schemas table was not made for this kind of use case. If you no longer have the clients data, it's no longer a valid schema and should not be listed on your schemas table. Silencing this exception does not sound like a good idea to me mate. Sorry, we can not add every feature.

**Comment by mikicz** [comment]
What do you think about the exception itelf if we disregard the ignore option? That does have use, right?

**Comment by bernardopires** [comment]
Oh yes, sorry, I think throwing a proper exception is definitely useful. I'll reopen this PR but alternatively you can send a new one. Please raise `MigrationSchemaMissing` instead of `RuntimeException`.

File: tenant_schemas/management/commands/migrate_schemas.py [hunk 1]
```diff
@@ -41,18 +42,23 @@ def handle(self, *args, **options):
...
             if self.schema_name and self.schema_name != self.PUBLIC_SCHEMA_NAME:
                 if not schema_exists(self.schema_name):
-                    raise RuntimeError('Schema "{}" does not exist'.format(
+                    raise MigrationSchemaMissing('Schema "{}" does not exist'.format(
                         self.schema_name))
                 else:
                     self.run_migrations(self.schema_name, settings.TENANT_APPS)
             else:
-                all_tenants = get_tenant_model().objects.exclude(schema_name=get_public_schema_name())
+                all_tenants = get_tenant_model().objects.exclude(schema_name=self.PUBLIC_SCHEMA_NAME)
```
Comment (bernardopires) [inline tenant_schemas/management/commands/migrate_schemas.py]: Can you please undo the change on this line?

**Comment by mikicz** [comment]
I merged master into the branch to get rid of the conflicts.

**Comment by bernardopires** [comment]
Thanks Mikuláš!
